analyses/org.eclipse.tracecompass.incubator.executioncomparision.core/src/org/eclipse/tracecompass/incubator/internal/executioncomparison/core/DifferentialSeqCallGraphAnalysis.java

```diff
@@ -0,0 +1,409 @@
+            WeightedTreeSet<ICallStackSymbol, Object> newTreeSet = new WeightedTreeSet<>();
+            String mainGroup = MERGE;
+
+            List<CallGraph> cGList = addToCallGraph(start, end, traceList);
+
+            for (CallGraph callGraph : cGList) {
+                Collection<ICallStackElement> elements = getLeafElements(callGraph);
+                for (ICallStackElement element : elements) {
+                    recurseAddElementData(element, mainGroup, callGraph, newTreeSet);
+                }
+            }
+            return newTreeSet;
+        }
+    }
+
+    /**
+     * Helper function that adds the call graphs for the specified time range
+     * and trace list to the call graph list.
+     *
+     * @param start
+     *            the start time of the time range
+     * @param end
+     *            the end time of the time range
+     * @param traceList
+     *            the list of trace names
+     * @return the list of call graphs
+     */
+    private static List<CallGraph> addToCallGraph(ITmfTimestamp start, ITmfTimestamp end, List<String> traceList) {
+        List<CallGraph> cGList = new ArrayList<>();
+        for (String traceName : traceList) {
+            ICallGraphProvider2 instrumentedCallStackAnalysis = fTraceCallGraphRegistry.get(traceName);
+            if (instrumentedCallStackAnalysis != null) {
+                ITmfTrace trace = getTrace(traceName);
+                if (trace != null) {
+                    ITmfTimestamp traceStart = start;
+                    ITmfTimestamp traceEnd = end;
+
+                    if (traceStart.getValue() < trace.getStartTime().getValue()) {
+                        traceStart = trace.getStartTime();
+                    }
+                    if (traceEnd.getValue() > trace.getEndTime().getValue()) {
+                        traceEnd = trace.getEndTime();
+                    }
+                    cGList.add(instrumentedCallStackAnalysis.getCallGraph(traceStart, traceEnd));
+                }
+            }
+        }
+        return cGList;
+    }
+
+    /**
+     * Gets the differential call graph
+     *
+     * @return DifferentialCallGraph
+     */
+    public IWeightedTreeSet<ICallStackSymbol, Object, DifferentialWeightedTree<ICallStackSymbol>> getCallGraph() {
+        DifferentialCallGraphProvider differentialCallGraphProvider = getDifferentialCallGraphProvider();
+        if (differentialCallGraphProvider != null) {
+            return differentialCallGraphProvider.getTreeSet();
+        }
+        return new DifferentialWeightedTreeSet<>();
+    }
+
+    /**
+     * Gets the differential weighted tree provider
+     *
+     * @param monitor
+     *            the progress monitor, can be null
+     * @return the differential weighted provider or null
+     */
+    public @Nullable DifferentialWeightedTreeProvider<?> getDiffProvider(@Nullable IProgressMonitor monitor) {
```

Comment:
Should not be nullable ?

Reply:
This method returns `fDifferentialCallGraphProvider` which is nullable. We could change `fDifferentialCallGraphProvider` to be Non-Nullable, but it would require changing the implementation a bit. I think the current implementation works fine, but we could definitely change it if you think its better.